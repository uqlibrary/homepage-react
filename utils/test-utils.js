/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Route } from 'react-router';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';

import preview from 'jest-preview';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';
import { createMemoryHistory } from 'history';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');

const { configure } = domTestingLib;

configure(config => ({
    ...config,
    testIdAttribute: 'id',
}));

const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <MuiPickersUtilsProvider utils={MomentUtils}>{props.children}</MuiPickersUtilsProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {},
) => {
    return render(
        <AllTheProviders>
            <Router history={history}>{ui}</Router>
        </AllTheProviders>,
    );
};

export const RenderWithRouter = ({
    children,
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
}) => (
    <AllTheProviders>
        <Router history={history}>{children}</Router>
    </AllTheProviders>
);
export const renderWithRedux = ({ initialState }) => render => {
    return {
        ...render,
        store: getStore({ initialState, history: render.history }),
    };
};

export const withRouter = ({
    route = '/',
    path = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
} = {}) => WrappedComponent => {
    return (
        <Router history={history}>
            <Route path={path} children={WrappedComponent} />
        </Router>
    );
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

// eslint-disable-next-line react/prop-types
export const WithRedux = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>{children}</Provider>
);

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    renderWithRouter,
    renderWithRedux,
    withRedux,
    withRouter,
    WithRedux,
    WithReduxStore,
    RenderWithRouter,
    preview,
};
