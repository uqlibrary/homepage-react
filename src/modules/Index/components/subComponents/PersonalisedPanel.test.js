import React from 'react';
import { rtlRender } from 'test-utils';
import PersonalisedPanel from './PersonalisedPanel';
function setup(testProps = {}) {
    const props = {
        account: { id: 'uqtest' },
        author: { aut_id: 79324 },
        loans: { recordNumber: 'vanilla' },
        printBalance: {
            balance: '12.50',
            cardNumber: '36122190182095876',
            email: 'k.lane@uq.edu.au',
            retrievedAt: '2020-11-25T11:50:15+00:00',
        },
        possibleRecords: 10,
        incompleteNTRORecords: { total: 18, took: 104, per_page: 20, current_page: 1, from: 1, to: 18 },
    };
    return rtlRender(<PersonalisedPanel {...props} {...testProps} />);
}

describe('PersonalisedPanel', () => {
    it('default values render correctly', () => {
        const { container } = setup();
        expect(container).toMatchSnapshot();
    });
});
