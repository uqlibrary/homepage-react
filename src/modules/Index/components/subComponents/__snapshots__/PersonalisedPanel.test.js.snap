// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalisedPanel default values render correctly 1`] = `
<div>
  <div
    class="flexWrapper css-1vniqbq"
    data-testid="personalised-panel"
    id="personalised-panel"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="flexHeader"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 greeting css-19cmg10-MuiTypography-root"
      >
        Good morning
         
      </h2>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
        style="margin-left: 16px; margin-top: 6px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-auto css-kjpdr9-MuiGrid-root"
        >
          <span
            aria-label="Your UQ username is uqtest"
            class="MuiTypography-root MuiTypography-body1 css-u1nwny-MuiTypography-root"
            data-mui-internal-clone-element="true"
            data-testid="pp-tooltip"
            style="font-size: 14px; color: rgb(89, 89, 89);"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall uqidIcon css-ptiqhd-MuiSvgIcon-root"
              data-testid="AccountBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-7 3c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m7 13H5v-.23c0-.62.28-1.2.76-1.58C7.47 15.82 9.64 15 12 15s4.53.82 6.24 2.19c.48.38.76.97.76 1.58z"
              />
            </svg>
            uqtest
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-true locationWrapper css-qixcmv-MuiGrid-root"
        >
          <div
            class="css-15tc0jm"
            data-testid="location"
            id="location"
          >
            <button
              aria-label="Click to update your preferred campus, currently not set"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary iconButton css-114ngwv-MuiButtonBase-root-MuiButton-root"
              data-analyticsid="location-button"
              data-mui-internal-clone-element="true"
              data-testid="location-button"
              id="location-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon wiggler css-i4bv87-MuiSvgIcon-root"
                data-testid="RoomIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
                />
              </svg>
               
              Set a preferred campus
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flexContent"
    />
    <div
      class="flexFooter"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        style="margin-left: 16px;"
      >
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
          style="min-height: 0px; width: 100%;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 menuItem css-1idn90j-MuiGrid-root"
              >
                <button
                  aria-label="Click to manage your print balance"
                  class="MuiButtonBase-root MuiButton-root menuItemRoot MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root menuItemRoot MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-1w2px6u-MuiButtonBase-root-MuiButton-root"
                  data-analyticsid="pp-papercut-tooltip"
                  data-mui-internal-clone-element="true"
                  data-testid="pp-papercut-menu-button"
                  id="pp-papercut-menu-button"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true menuItemLabel css-1vd824g-MuiGrid-root"
                      style="line-height: 1.2; padding-right: 0px;"
                    >
                      Manage your print balance ($12.50)
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
                    >
                      <div
                        class="itemButton"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                          data-testid="PrintIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 8H5c-1.66 0-3 1.34-3 3v6h4v4h12v-4h4v-6c0-1.66-1.34-3-3-3m-3 11H8v-5h8zm3-7c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1m-1-9H6v4h12z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
          style="min-height: 0px; width: 100%;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 menuItemAnchor css-1idn90j-MuiGrid-root"
              >
                <a
                  aria-label="Click to manage your library loans"
                  class="menuItemRootAnchor"
                  data-analyticsid="pp-loans-tooltip"
                  data-mui-internal-clone-element="true"
                  data-testid="pp-loans-menu-button"
                  href="https://search.library.uq.edu.au/primo-explore/account?vid=61UQ&section=loans&lang=en_US"
                  id="pp-loans-menu-button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true menuItemLabel css-1vd824g-MuiGrid-root"
                      >
                        Manage your library loans 
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto anchorBadge css-1wrgmsj-MuiGrid-root"
                      >
                        <div
                          class="itemButton"
                        >
                          <span
                            class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                              data-testid="MenuBookIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M21 5c-1.11-.35-2.33-.5-3.5-.5-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5V6c-.6-.45-1.25-.75-2-1m0 13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5z"
                              />
                              <path
                                d="M17.5 10.5c.88 0 1.73.09 2.5.26V9.24c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99M13 12.49v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26V11.9c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.3-4.5.83m4.5 1.84c-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26v-1.52c-.79-.16-1.64-.24-2.5-.24"
                              />
                            </svg>
                            <span
                              class="MuiBadge-badge ppBadgeInfo MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-13ffwhj-MuiBadge-badge"
                            />
                          </span>
                        </div>
                      </div>
                    </div>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px; width: 100%;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 menuItemAnchor css-1idn90j-MuiGrid-root"
              >
                <a
                  aria-label="Click to manage your library fines"
                  class="menuItemRootAnchor"
                  data-analyticsid="pp-fines-tooltip"
                  data-mui-internal-clone-element="true"
                  data-testid="pp-fines-menu-button"
                  href="https://search.library.uq.edu.au/primo-explore/account?vid=61UQ&section=fines&lang=en_US"
                  id="pp-fines-menu-button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true menuItemLabel css-1vd824g-MuiGrid-root"
                      >
                        Manage your library fines ($undefined)
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto anchorBadge css-1wrgmsj-MuiGrid-root"
                      >
                        <div
                          class="itemButton"
                        >
                          <span
                            class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                              data-testid="MonetizationOnIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1.41 16.09V20h-2.67v-1.93c-1.71-.36-3.16-1.46-3.27-3.4h1.96c.1 1.05.82 1.87 2.65 1.87 1.96 0 2.4-.98 2.4-1.59 0-.83-.44-1.61-2.67-2.14-2.48-.6-4.18-1.62-4.18-3.67 0-1.72 1.39-2.84 3.11-3.21V4h2.67v1.95c1.86.45 2.79 1.86 2.85 3.39H14.3c-.05-1.11-.64-1.87-2.22-1.87-1.5 0-2.4.68-2.4 1.64 0 .84.65 1.39 2.67 1.91s4.18 1.39 4.18 3.91c-.01 1.83-1.38 2.83-3.12 3.16"
                              />
                            </svg>
                            <span
                              class="MuiBadge-badge ppBadgeError MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-13ffwhj-MuiBadge-badge"
                            />
                          </span>
                        </div>
                      </div>
                    </div>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
          style="min-height: 0px; width: 100%;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 menuItemAnchor css-1idn90j-MuiGrid-root"
              >
                <a
                  aria-label="Click to visit your UQ eSpace dashboard"
                  class="menuItemRootAnchor"
                  data-analyticsid="pp-espace-possible-tooltip"
                  data-mui-internal-clone-element="true"
                  data-testid="pp-espace-possible-menu-button"
                  href="https://espace.library.uq.edu.au/records/possible"
                  id="pp-espace-possible-menu-button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true menuItemLabel css-1vd824g-MuiGrid-root"
                      >
                        Claim 10 UQ eSpace records
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto anchorBadge css-1wrgmsj-MuiGrid-root"
                      >
                        <div
                          class="itemButton"
                        >
                          <span
                            class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                              data-testid="AssignmentIndIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1m0 4c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3m6 12H6v-1.4c0-2 4-3.1 6-3.1s6 1.1 6 3.1z"
                              />
                            </svg>
                            <span
                              class="MuiBadge-badge ppBadgeInfo MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-1df0kgp-MuiBadge-badge"
                            >
                              10
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
          style="min-height: 0px; width: 100%;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 menuItemAnchor css-1idn90j-MuiGrid-root"
              >
                <a
                  aria-label="Click to visit UQ eSpace to link/create your ORCID account"
                  class="menuItemRootAnchor"
                  data-analyticsid="pp-espace-orcid-tooltip"
                  data-mui-internal-clone-element="true"
                  data-testid="pp-espace-orcid-menu-button"
                  href="https://espace.library.uq.edu.au/author-identifiers/orcid/link"
                  id="pp-espace-orcid-menu-button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true menuItemLabel css-1vd824g-MuiGrid-root"
                      >
                        Link ORCiD account to UQ eSpace
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto anchorBadge css-1wrgmsj-MuiGrid-root"
                      >
                        <div
                          class="itemButton"
                        >
                          <span
                            class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                              data-testid="LinkIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5"
                              />
                            </svg>
                            <span
                              class="MuiBadge-badge ppBadgeWarning MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-1df0kgp-MuiBadge-badge"
                            >
                              !
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
          style="min-height: 0px; width: 100%;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 menuItemAnchor css-1idn90j-MuiGrid-root"
              >
                <a
                  aria-label="Click to visit UQ eSpace to complete these NTRO records"
                  class="menuItemRootAnchor"
                  data-analyticsid="pp-espace-ntro-tooltip"
                  data-mui-internal-clone-element="true"
                  data-testid="pp-espace-ntro-menu-button"
                  href="https://espace.library.uq.edu.au/records/incomplete"
                  id="pp-espace-ntro-menu-button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true menuItemLabel css-1vd824g-MuiGrid-root"
                      >
                        Complete 18 NTRO records in UQ eSpace
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto anchorBadge css-1wrgmsj-MuiGrid-root"
                      >
                        <div
                          class="itemButton"
                        >
                          <span
                            class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                              data-testid="PostAddIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M17 19.22H5V7h7V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-7h-2z"
                              />
                              <path
                                d="M19 2h-2v3h-3c.01.01 0 2 0 2h3v2.99c.01.01 2 0 2 0V7h3V5h-3zM7 9h8v2H7zm0 3v2h8v-2h-3zm0 3h8v2H7z"
                              />
                            </svg>
                            <span
                              class="MuiBadge-badge ppBadgeWarning MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-1df0kgp-MuiBadge-badge"
                            >
                              18
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
